fuse_constraints/src/relative_pose_2d_stamped_constraint.cpp

```diff
@@ -45,12 +45,58 @@ RelativePose2DStampedConstraint::RelativePose2DStampedConstraint(
   const fuse_variables::Orientation2DStamped& orientation1,
   const fuse_variables::Position2DStamped& position2,
   const fuse_variables::Orientation2DStamped& orientation2,
-  const fuse_core::Vector3d& delta,
-  const fuse_core::Matrix3d& covariance) :
-    fuse_core::Constraint{position1.uuid(), orientation1.uuid(), position2.uuid(), orientation2.uuid()},
-    delta_(delta),
-    sqrt_information_(covariance.inverse().llt().matrixU())
+  const fuse_core::Vector3d& partial_delta,
+  const fuse_core::Matrix3d& partial_covariance,
+  const std::vector<size_t>& linear_indices,
+  const std::vector<size_t>& angular_indices) :
+    fuse_core::Constraint{position1.uuid(), orientation1.uuid(), position2.uuid(), orientation2.uuid()}
 {
+  size_t total_variable_size = position1.size() + orientation1.size();
+  size_t total_indices = linear_indices.size() + angular_indices.size();
+
+  assert(partial_delta.rows() == static_cast<int>(total_indices));
+  assert(partial_covariance.rows() == static_cast<int>(total_indices));
+  assert(partial_covariance.cols() == static_cast<int>(total_indices));
+
+   // Compute the sqrt information of the provided cov matrix
+  fuse_core::MatrixXd partial_sqrt_information = partial_covariance.inverse().llt().matrixU();
+
+   // Assemble a mean vector and sqrt information matrix from the provided values, but in proper variable order
+  // What are we doing here?
+  // The constraint equation is defined as: cost(x) = ||A * (x - b)||^2
+  // If we are measuring a subset of dimensions, we only want to produce costs for the measured dimensions.
+  // But the variable vectors will be full sized. We can make this all work out by creating a non-square A
+  // matrix, where each row computes a cost for one measured dimensions, and the columns are in the order
+  // defined by the variable.
+  delta_ = fuse_core::VectorXd::Zero(total_variable_size);
```

Comment:
`delta_` is a fixed-sized vector, `Eigen::Vector3d`. Should change this to:
```suggestion
  delta_ = fuse_core::Vector3d::Zero();
```

or simply
```suggestion
  delta_.setZero();
```